File: 62-UniquePaths/62-UniquePaths.py
# Last updated: 9/16/2025, 12:10:04 AM
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        # Create a 1D array 'dp' with all elements initialized to 1
        dp = [1] * n
    
        # Fill the dp array for each row
        for i in range(1, m):
            for j in range(1, n):
                dp[j] += dp[j-1]

        return dp[n-1]        

File: 62-UniquePaths/62-UniquePaths.cpp
// Last updated: 9/16/2025, 12:10:00 AM
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>>dp(m,vector<int>(n,0));
        for(int i=0;i<n;i++){
            dp[0][i]=1;
        }
        for(int j=0;j<m;j++){
            dp[j][0]=1;
        }

        for(int i=1;i<m;i++){
            for(int j=1;j<n;j++){
                int down=dp[i-1][j];
                int right=dp[i][j-1];
                dp[i][j]=down+right;
            }
        }
        return dp[m-1][n-1];
    }
};